python_modules/dagster/dagster/components/lib/sql_component/sql_component.py

```diff
@@ -1,39 +1,51 @@
 from abc import ABC, abstractmethod
 from collections.abc import Iterable, Mapping
 from pathlib import Path
-from typing import Annotated, Any, Generic, Optional, Union
+from typing import Annotated, Any, Optional, Union
 
 from dagster_shared import check
-from jinja2 import Template
 from pydantic import BaseModel, Field
-from typing_extensions import TypeVar
 
+from dagster._annotations import preview, public
 from dagster._core.definitions.result import MaterializeResult
 from dagster._core.execution.context.asset_check_execution_context import AssetCheckExecutionContext
 from dagster._core.execution.context.asset_execution_context import AssetExecutionContext
 from dagster.components.core.context import ComponentLoadContext
 from dagster.components.lib.executable_component.component import ExecutableComponent
+from dagster.components.lib.sql_component.sql_client import SQLClient
+from dagster.components.resolved.base import Resolvable
 from dagster.components.resolved.core_models import OpSpec
-from dagster.components.resolved.model import Model, Resolver
-
-T = TypeVar("T")
-
-
-class SqlComponent(ExecutableComponent, Model, BaseModel, Generic[T], ABC):
-    """Base component which executes templated SQL."""
-
-    execution: Optional[OpSpec] = None
+from dagster.components.resolved.model import Resolver
+
+
+@public
+@preview
+class SqlComponent(ExecutableComponent, Resolvable, ABC):
+    """Base component which executes templated SQL. Subclasses
+    implement instructions on where to load the SQL content from
+    and how to execute it.
+    """
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    connection: Annotated[
+        Annotated[
+            SQLClient,
+            Resolver(lambda ctx, value: value, model_field_type=Any),  # type: ignore
```

Comment:
What's the deal with this / why not `Resolver.default()`? Realistically this is doing the `Resolver.passthrough()` thing, but it seems like we'd want the resolver to be invoked.

Reply:
Also if we only support passing by reference at the moment, should the model_field_type just be `str`?